specification/maintenance/Maintenance.Management/MaintenanceConfiguration.tsp

```diff
@@ -0,0 +1,151 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Maintenance;
+/**
+ * Maintenance configuration record type
+ */
+@subscriptionResource
+model MaintenanceConfiguration
+  is Azure.ResourceManager.TrackedResource<MaintenanceConfigurationProperties> {
+  ...ResourceNameParameter<
+    Resource = MaintenanceConfiguration,
+    KeyName = "resourceName",
+    SegmentName = "publicMaintenanceConfigurations",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface MaintenanceConfigurationOps
+  extends Azure.ResourceManager.Legacy.CustomLegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = MaintenanceConfiguration,
+        KeyName = "resourceName",
+        SegmentName = "publicMaintenanceConfigurations",
+        NamePattern = ""
+      >>,
+      MaintenanceError
+    > {}
+
+@armResourceOperations
+interface MaintenanceConfigurations {
+  /**
+   * Get Public Maintenance Configuration record
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("PublicMaintenanceConfigurations_Get")
+  get is MaintenanceConfigurationOps.Read<MaintenanceConfiguration>;
+
+  /**
+   * Get Public Maintenance Configuration records
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @segment("publicMaintenanceConfigurations")
+  @operationId("PublicMaintenanceConfigurations_List")
+  list is MaintenanceConfigurationOps.List<
+    MaintenanceConfiguration,
+    Response = ArmResponse<ListMaintenanceConfigurationsResult>
+  >;
+}
+@armResourceOperations
+interface MaintenanceConfigurationOperationGroupOps
+  extends Azure.ResourceManager.Legacy.CustomLegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = MaintenanceConfiguration,
+        KeyName = "resourceName",
+        SegmentName = "maintenanceConfigurations",
+        NamePattern = ""
+      >>,
+      MaintenanceError
+    > {}
+
+@armResourceOperations
+interface MaintenanceConfigurationOperationGroup {
```

Comment:
Why not use the appropriate name?

Reply:
Note that I would prefer using the same interface and *not* using `@operationId`  over splitting things into two interfaces